### PR TITLE
Adds voice assistant announce

### PR DESCRIPTION
**Pull request by synesthesiam**
Adds `VoiceAssistantAnnounce` message and `send_voice_assistant_announce`

* https://github.com/esphome/esphome/pull/7377
* https://github.com/home-assistant/core/pull/125157
* https://github.com/esphome/voice-kit/pull/64

### CODE REVIEW

**Comment by synesthesiam** [comment]
Leaving as a draft since this is still under discussion.

**Comment by bdraco** [comment]
I fixed codecov so it should report correctly now

File: aioesphomeapi/client.py [hunk 1]
```diff
@@ -1286,6 +1289,13 @@ def subscribe_voice_assistant(
             ]
             | None
         ) = None,
+        handle_announce_finished: (
```
Comment (jesserockz) [inline aioesphomeapi/client.py:1292]: Wouldn't it be better to have this callback in the `send_voice_assistant_announce` method?

Comment (synesthesiam) [reply aioesphomeapi/client.py:1292]: That should work fine, and would make the HA side simpler.

Comment (synesthesiam) [reply aioesphomeapi/client.py]: Moved the wait inside aioesphomeapi

File: aioesphomeapi/model.py [hunk 2]
```diff
@@ -1291,6 +1292,11 @@ class VoiceAssistantAudioData(APIModelBase):
     end: bool = False
 
 
+@_frozen_dataclass_decorator
+class VoiceAssistantAnnounceFinished(APIModelBase):
+    success: bool
```
[the remark this answers is not in the document]
Comment (jesserockz) [reply aioesphomeapi/model.py:1297]: We always define the default value here. protobuf removes the values from the message if it is the default. For example if you sent a success=false message back, its only gonna send the message id to save on bytes.

**Review by jesserockz — APPROVED** [review]

**Pull request merged**